docs: add surfpool configuration reference to Anchor.toml docs (#4479)
otter-sec/anchor: docs/content/docs/references/anchor-toml.mdx

filename = "some_other_account.json"
```

## surfpool

> Configuration for the [Surfpool](https://github.com/txtx/surfpool) validator,

Reviewer: ```suggestion
Configuration for the [Surfpool](https://github.com/solana-foundation/surfpool) validator,
```

which is the default local validator used by `anchor test`. Surfpool provides a
lightweight Solana simulator with features like automatic account cloning and
runbook execution. All fields are optional and have sensible defaults.

Example:

```toml
[surfpool]
startup_wait = 5000                                       # Time (ms) to wait for Surfpool to start. Default: 5000.
shutdown_wait = 2000                                      # Time (ms) to wait for Surfpool to shut down. Default: 2000.
rpc_port = 8899                                           # JSON RPC port. Default: 8899.
ws_port = 8900                                            # WebSocket port. If not set, derived automatically.
host = "127.0.0.1"                                        # IP address to bind to. Default: "127.0.0.1".
online = true                                             # Enable online mode (clone accounts from a remote cluster).

> datasource_rpc_url = "https://api.mainnet-beta.solana.com" # RPC URL to use as the data source when online mode is enabled.

Reviewer: ```suggestion
datasource_rpc_url = "https://api.mainnet.solana.com" # RPC URL to use as the data source when online mode is enabled.
```

slot_time = 400                                            # Simulated slot time in milliseconds.
log_level = "info"                                         # Log level for Surfpool. Default: "none".
block_production_mode = "clock"                            # Block production mode. Default: "transaction".
```

### startup_wait

Time in milliseconds to wait for Surfpool to start up. This is useful when
loading many accounts or running runbooks at startup. Default: `5000`.

### shutdown_wait

Time in milliseconds to wait for Surfpool to shut down gracefully.
Default: `2000`.

### rpc_port

The port on which Surfpool exposes its JSON RPC endpoint. Default: `8899`.

### ws_port

The port for the WebSocket endpoint. If not specified, it is derived
automatically.

### host

The IP address to bind the Surfpool validator ports to. Default: `"127.0.0.1"`.

### online

When set to `true`, Surfpool operates in online mode, allowing it to clone
accounts from a remote cluster specified by `datasource_rpc_url`. Default:
`false` (offline mode).

### datasource_rpc_url

The RPC URL of the remote cluster used as a data source when `online` is
enabled.

Example:

```toml
[surfpool]
online = true

> datasource_rpc_url = "https://api.mainnet-beta.solana.com"

Reviewer: ```suggestion
datasource_rpc_url = "https://api.mainnet.solana.com"
```
